setup.py

```diff
@@ -190,83 +161,14 @@
     nbconvert=["nbconvert"],
 )
 
-install_requires = [
-    "setuptools>=18.5",
-    "jedi>=0.16",
-    "decorator",
-    "pickleshare",
-    "traitlets>=4.2",
-    "prompt_toolkit>=2.0.0,<3.1.0,!=3.0.0,!=3.0.1",
-    "pygments",
-    "backcall",
-    "stack_data",
-    "matplotlib-inline",
-]
-
-# Platform-specific dependencies:
-# This is the correct way to specify these,
-# but requires pip >= 6. pip < 6 ignores these.
-
-extras_require.update(
-    {
-        ':sys_platform != "win32"': ["pexpect>4.3"],
-        ':sys_platform == "darwin"': ["appnope"],
-        ':sys_platform == "win32"': ["colorama"],
-    }
-)
-# FIXME: re-specify above platform dependencies for pip < 6
-# These would result in non-portable bdists.
-if not any(arg.startswith('bdist') for arg in sys.argv):
-    if sys.platform == 'darwin':
-        install_requires.extend(['appnope'])
-
-    if not sys.platform.startswith("win"):
-        install_requires.append("pexpect>4.3")
-
-    # workaround pypa/setuptools#147, where setuptools misspells
-    # platform_python_implementation as python_implementation
-    if 'setuptools' in sys.modules:
-        for key in list(extras_require):
-            if 'platform_python_implementation' in key:
-                new_key = key.replace('platform_python_implementation', 'python_implementation')
-                extras_require[new_key] = extras_require.pop(key)
-
-everything = set()
-for key, deps in extras_require.items():
-    if ':' not in key:
-        everything.update(deps)
-extras_require['all'] = list(sorted(everything))
-
-if 'setuptools' in sys.modules:
-    setuptools_extra_args['python_requires'] = '>=3.7'
-    setuptools_extra_args['zip_safe'] = False
-    setuptools_extra_args['entry_points'] = {
-        'console_scripts': find_entry_points(),
-        'pygments.lexers': [
-            'ipythonconsole = IPython.lib.lexers:IPythonConsoleLexer',
-            'ipython = IPython.lib.lexers:IPythonLexer',
-            'ipython3 = IPython.lib.lexers:IPython3Lexer',
-        ],
-    }
-    setup_args['extras_require'] = extras_require
-    setup_args['install_requires'] = install_requires
 
-else:
-    # scripts has to be a non-empty list, or install_scripts isn't called
-    setup_args['scripts'] = [e.split('=')[0].strip() for e in find_entry_points()]
+everything = set(chain.from_iterable(extras_require.values()))
+extras_require['all'] = list(sorted(everything))
 
-    setup_args['cmdclass']['build_scripts'] = build_scripts_entrypt
+setup_args['extras_require'] = extras_require
 
 #---------------------------------------------------------------------------
 # Do the actual setup now
 #---------------------------------------------------------------------------
 
-setup_args.update(setuptools_extra_args)
-
-
-
-def main():
-    setup(**setup_args)
-
-if __name__ == '__main__':
-    main()
+setup(**setup_args)
```

Comment:
This is a footgun for pytest, is removing the guard necessary?

Reply:
I suppose it isn't necessary and I wasn't aware of any interaction with pytest. Why would pytest be testing `setup.py` though?

Reply:
doctest plugin loads every python script to collect and test examples in documentation